#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_build_fix.dpatch by Anthony Fok <foka@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from SuSE's Mike Fabian <mfabian@suse.de> to fix build.

@DPATCH@
diff -urNad libfont-freetype-perl-0.03~/FreeType.xs libfont-freetype-perl-0.03/FreeType.xs
--- libfont-freetype-perl-0.03~/FreeType.xs	2008-07-27 15:22:06.000000000 +0800
+++ libfont-freetype-perl-0.03/FreeType.xs	2008-07-27 15:23:14.000000000 +0800
@@ -21,6 +21,7 @@
 #include FT_GLYPH_H
 #include FT_OUTLINE_H
 #include FT_BBOX_H
+#include FT_TYPE1_TABLES_H
 
 #undef assert
 #include <assert.h>
@@ -805,7 +806,8 @@
             char_code = FT_Get_First_Char(face, &glyph_idx);
             while (glyph_idx) {
                 if (glyph_idx == glyph->index) {
-                    RETVAL = newSVuv((UV) glyph->char_code = char_code);
+                    glyph->char_code = char_code;	
+                    RETVAL = newSVuv((UV) glyph->char_code);
                     break;
                 }
                 char_code = FT_Get_Next_Char(face, char_code, &glyph_idx);
@@ -1052,7 +1054,7 @@
         rows = newAV();
         av_extend(rows, bitmap->rows - 1);
         buf = bitmap->buffer;
-        row_buf = New(0, row_buf, bitmap->width, unsigned char);
+        New(0, row_buf, bitmap->width, unsigned char);
 
         if (bitmap->pixel_mode == FT_PIXEL_MODE_MONO) {
             for (i = 0; i < bitmap->rows; ++i) {
